docs/user-guide/multiple_sources.md

```diff
@@ -0,0 +1,62 @@
+# Multiple Sources for an Application
+
+Argo CD has the ability to specify multiple sources to add services to the Application. Argo CD compiles all the sources and reconciles each source individually for creating the application.
+
+You can provide multiple sources using the `sources` field. When you specify the sources field, Argo CD will ignore the values under `source` field for generating the application.
+
+See below example for specifying multiple sources:
+
+```
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: guestbook
+  namespace: argocd
+  labels:
+    argocd.argoproj.io/refresh: hard
+spec:
+  project: default
+  syncPolicy:
+    automated:
+      prune: true
+      selfHeal: true
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: argocd
+  sources:
+    - chart: elasticsearch
+      helm:
+        valueFiles:
+          - values.yaml
+      repoURL: https://helm.elastic.co
+      targetRevision: 7.6.0
+    - repoURL: https://github.com/argoproj/argocd-example-apps.git
+      path: guestbook
+      targetRevision: HEAD
+```
+
+The above example has 2 sources specified. Argo CD will reconcile each source separately and combine the resources that are generated for generating the application.
+
+In case application has multiple entries for the same source (repoURL), Argo CD would pick the source that is mentioned later in the list of sources. For example, consider the below list of sources:
+
+```
+sources:
+    - chart: elasticsearch
+      helm:
+        valueFiles:
+          - values.yaml
+      repoURL: https://helm.elastic.co
+      targetRevision: 7.6.0
+    - repoURL: https://github.com/argoproj/argocd-example-apps.git
+      path: guestbook
+      targetRevision: HEAD
```

Comment:
What happens if I have another source from this same repoURL but with a different path and/or targetRevision?

Reply:
Same question with using  externalValueFiles/repoURL on top:
- does the repoURL uniqueness check applies only for sources/repoURL?
- what if for one source i have multiple externalValueFiles pointing to the same repoURL, but different versions, different files?
- what if i have 2 sources pointing to the same repoURL but using different valueFiles or externalValueFiles?

Do we keep a kind of (repo, chart/path) couple for duplicate checks? 

Edit:
Just remembered the "externalValueFiles" was reworked with using reference to a sources/repoURL.
But I guess the question remains.